src/ripple/crypto/impl/RFC1751.cpp

```diff
@@ -393,25 +393,22 @@ int RFC1751::wsrch (std::string const& strWord, int iMin, int iMax)
 //        -2 words OK but parity is wrong
 int RFC1751::etob (std::string& strData, std::vector<std::string> vsHuman)
 {
-    int i, p, v, l;
-    char b[9];
-
     if (6 != vsHuman.size ())
         return -1;
 
-    memset (b, 0, sizeof (b));
+    int i, p = 0;
+    char b[9] = { 0 };
```

Comment:
I'd write `char b[9]{};` for things like this (it's fine as-is, I'm just not used to seeing `= {0}`.

Reply:
I think there's a difference between what `{}` and `{ 0 }` produce.  I should test this before writing this comment of course, but if I'm remembering correctly...

- `{}` will default initialize all nine members of the array, and
- `{ 0 }` will only initialize the first member of the array.

So, if that's correct, `{}` is more conservative and potentially better for long term maintenance.  For example, if you initialize only the first member of the `char` array, then it _might_ be undefined behavior if someone reads an element of the array other than the first.

Explaining the _might_: I'm pretty sure it would not be undefined for an array of `std::byte` or `unsigned char`; it's possible that `char` is okay?  I don't have a copy of the standard with me right now...

But `{ 0 }` is less expensive if we're counting cycles.

Reply:
I spoke with Scott and Howard this morning, and `char b[9] = {0};` and `char b[9]{}` both do the same thing, so it is fine as-is. The rule is that the elements without an initial value will be zero initialized, so `char b[9] = {1,2}` sets the first two elements and the rest will be zero. However, I do think this is a more obscure rule than `char b[9]{};`. Neither Scott S. nor I knew that off the top of our heads (Howard did of course :-)

Reply:
That's interesting. I thought that this `b[9]{}` was more obscure; I knew that would "default-initialize" but I didn't know off-hand if "default" in this case means 0, but I knew that `b[9] = { /* something */ };` initializes the first element to "something" and zero-initializes the rest.

I'm fine with doing `char b[9]{}`